FEXCore/Source/Interface/Core/OpcodeDispatcher.cpp

```diff
@@ -1740,6 +1743,11 @@ void OpDispatchBuilder::RotateOp(OpcodeArgs) {
     GetSrcSize(Op) == OpSize::i32Bit ? std::make_optional(&OpDispatchBuilder::ZeroShiftResult) : std::nullopt);
 }
 
+template<bool Left, bool IsImmediate, bool Is1Bit>
```

Comment:
Is there a long term reason to have all these 'thunk templates', or is this just short term to avoid rewriting all the call sites?

Reply:
The dispatcher uses function pointer tables that must be filled with functions of the same signature. The new functions have extra parameters not compatible with that signature, whereas the templates allow "hiding" these parameters such that the function pointers can be written to the table.

We could instead hide the extra parameters as state somewhere (ew) so they don't need to be passed directly to the functions. In comparison the template wrappers seem like the lesser evil for now :)

Reply:
Ah, gotcha, ok. That seems reasonable then (and indeed the lesser evil given the rest of the design this slots into ... the whole structure of the dispatcher indeed assumes templates, a C project would probably structure things differently to the same effect.) Anyway, seems reasonable to me, thanks for the explanation 👍